### DISCUSSION TITLE
How to disconnect?

### DISCUSSION THREAD

**Discussion by KubbusYC** [Discord4J/Discord4J#845] [Q&A] [answered]
Heya

I don't really know where to ask this, so I figured might as well give it a shot here.

I cannot for the life of me figure out how to make my bot leave the darn voice channel it is in, as your tutorial conveniently leaves that part out and the only code I've managed to find that successfully disconnect from voice are from back in 2.x.x.

Please, someone take pity upon my noobedness.

Help

**Comment by KubbusYC** [comment]
I mean I've figured out how to just make the bot crash anytime anyone sends any type of text input to any text channel. But like I don't know man.

**Comment by KubbusYC** [comment, marked as answer]
Aight, got it

So instead of setting up an extra leave command, i just checked within the join command, whether any message in the text channel equals the leave command, allowing me to reference the connection variable made when joining by virtue of being within the same function.

just be sure not to do that in a while loop, you'll open all the threads your cpu can do without an killing itself horribly

ok bye

**Reply by quanticc** [reply]
Hey there. sorry I missed this question and noticed that GitHub didn't notify me

Glad to know you solved it, but I'm adding this example code for future reference:

```java
    public static class Leave extends EventHandler {

        private final GatewayDiscordClient client;

        public Leave(GatewayDiscordClient client) {
            this.client = client;
        }

        @Override
        public Mono<Void> onMessageCreate(MessageCreateEvent event) {
            if (event.getMessage().getContent().equals("!vc leave")) {
                return Mono.justOrEmpty(event.getMember())
                        .flatMap(Member::getVoiceState)
                        .flatMap(vs -> client.getVoiceConnectionRegistry()
                                .getVoiceConnection(vs.getGuildId())
                                .doOnSuccess(vc -> {
                                    if (vc == null) {
                                        log.info("No voice connection to leave!");
                                    }
                                }))
                        .flatMap(VoiceConnection::disconnect);
            }
            return Mono.empty();
        }
    }
```
